handlers/payments.go

```diff
@@ -163,3 +164,57 @@ func HandlePatchPaymentSession(w http.ResponseWriter, req *http.Request) {
 
 	log.InfoR(req, "Successful PATCH request for payment resource", log.Data{"payment_id": paymentSession.MetaData.ID, "status": http.StatusOK})
 }
+
+// HandleGetPaymentDetails retrieves the payment details from the external provider
+func HandleGetPaymentDetails(w http.ResponseWriter, req *http.Request) {
+	// Get the payment session
+	vars := mux.Vars(req)
+	id := vars["payment_id"]
+	if id == "" {
+		log.ErrorR(req, fmt.Errorf("payment id not supplied"))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// The payment session must be retrieved directly to enable access to metadata outside the data block
+	paymentSession, _, err := paymentService.GetPaymentSession(req, id)
```

Comment:
this can be reworked to get the payment session from request context if we use the `pa.PaymentAuthenticationIntercept` as mentioned in previous comment